### ISSUE TITLE
Reference to varargs is wrong at call sites

### ISSUE THREAD

**Issue by panacekcz** [typetools/checker-framework#1635]
When an annotation refers to a varargs parameter such as `String get(@IndexFor("#2") int i, String... varargs)`, at call sites, this reference is resolved to the second argument (first element of the varargs array), not the array itself.

This issue can probably only show when using the index checker, but it seems to be a general problem of the framework (in `FlowExpressionParseUtil.parseParameter`).

Example code:
```java
import org.checkerframework.checker.index.qual.IndexFor;

public class IndexForVarargs {
	String get(@IndexFor("#2") int i, String... varargs) {
		return varargs[i];
	}
	
	void m() {
		get(1, "a", "b"); // line 9
		get(2, "abc"); // line 10
	}
}
```
Expected output:
- no error at line 9
- upper bound error at line 10

Run the checker as:
```
checker/bin/javac -processor org.checkerframework.checker.index.IndexChecker IndexForVarargs.java
```

Output:
```
IndexForVarargs.java:9: error: [argument.type.incompatible] incompatible types in argument.
		get(1, "a", "b");
		    ^
  found   : @UpperBoundUnknown int
  required: @LTLengthOf(value=""a"", offset={}) int
1 error
```

**Comment by smillst** [comment]
At the call `get(1, "a", "b")`, 
`@IndexFor(#1)` should translate to `@LTLengthOf("new String[] {"abc"}")`.  I can implement that.  However, `FlowExpressionParseUtil#parse` can't parse array creations, so `ValueAnnotatedTypeFactory#getMinLenFromString` still returns 0.  

Updating `FlowExpressionParseUtil#parse` is the right way to fix this, but we want to replace it with a real parser (#1614). 

In the mean time, I could probably implement a work around for the Upper Bound Checker.
